daemons/pacemakerd/pacemakerd.c

```diff
@@ -241,9 +277,12 @@ main(int argc, char **argv)
         goto done;
     }
 
+    pcmk__force_args(context, &error, "%s --xml-simple-list --xml-substitute", g_get_prgname());
```

Comment:
I see we tend to force args everywhere now. I thought that was originally to match pre-existing output as closely as possible -- is it useful for new XML output as well?

Reply:
That was the original intention.  With the code as it is right now, the output from this will look like:
```<pacemaker-result api-version="2.3" request="/home/clumens/src/pacemaker/daemons/pacemakerd/.libs/pacemakerd -F --output-as xml">
  <pacemakerd version="2.1.0" build="3bf47be3f" feature_set="3.10.0">
    <features>
      <feature>agent-manpages</feature>
      <feature>ascii-docs</feature>
      ...
    </features>
  </pacemakerd>
  <status code="0" message="OK"/>
</pacemaker-result>
```

If I remove the call to pcmk__force_args but keep pacemaked_features_xml as it is (using pcmk__output_create_xml_text_node instead of out->list_item), the output will look like this:
```<pacemaker-result api-version="2.3" request="/home/clumens/src/pacemaker/daemons/pacemakerd/.libs/pacemakerd -F --output-as xml">
  <pacemakerd version="2.1.0" build="3bf47be3f" feature_set="3.10.0">
    <list name="features" count="10">
      <feature>agent-manpages</feature>
      <feature>ascii-docs</feature>
      ...
    </list>
  </pacemakerd>
  <status code="0" message="OK"/>
</pacemaker-result>
```

And then if I also use out->list_item and go all the way with the generic list code, it'd look like this:
```<pacemaker-result api-version="2.3" request="/home/clumens/src/pacemaker/daemons/pacemakerd/.libs/pacemakerd -F --output-as xml">
  <pacemakerd version="2.1.0" build="3bf47be3f" feature_set="3.10.0">
    <list name="features" count="10">
      <item name="feature">agent-manpages</item>
      <item name="feature">ascii-docs</item>
      ...
    </list>
  </pacemakerd>
  <status code="0" message="OK"/>
</pacemaker-result>
```

Reply:
I'm ambivalent. The forced way certainly looks better, but we'll deal with the code more than the XML, so the simpler code for the generic way is appealing. If we ever might want to give some extended info for the features, the structured way makes sense (maybe e.g. <feature value="1">default-resource-stickiness</feature>), but that's a stretch, it will probably always be a simple list. I'm fine with whatever you decide.

Reply:
I feel like the forced way also looks the most XML-ish, and is maybe what we should just be doing everywhere all the time.  I don't know that we are actually using the generic list anywhere.  It's referenced in crm_resource*.rng so I suppose one thing there might be using it.

Reply:
Regardless, we don't need --xml-substitute here.  Nuked that.